src/app/modules/feedback/feedback.service.ts

```diff
@@ -151,3 +157,103 @@ export const hasNoPreviousFeedback = (
     }
     return okAsync(true as const)
   })
+
+/**
+ * Inserts given admin feedback to the database.
+ * @param userId the userId of the admin that provided the feedback
+ * @param rating the feedback rating to insert (0 for thumbs down, 1 for thumbs up)
+ * @param comment the feedback comment to insert if available
+ * @returns ok(IAdminFeedbackSchema) if successfully inserted
+ * @returns err(DatabaseError) on database error
+ */
+export const insertAdminFeedback = ({
+  userId,
+  rating,
+  comment,
+}: {
+  userId: string
+  rating: number
+  comment?: string
+}) => {
+  return ResultAsync.fromPromise(
+    AdminFeedbackModel.create({
+      userId,
+      rating,
+      comment,
+    }),
+    (error) => {
+      logger.error({
+        message: 'Database error when creating admin feedback document',
+        meta: {
+          action: 'insertAdminFeedback',
+          userId,
+        },
+        error,
+      })
+
+      return new DatabaseError('Admin feedback could not be created')
+    },
+  )
+}
+
+/**
+ * Updates admin feedback in the database.
+ * Will use previous value if comment or rating are not passed into method
+ * @param feedbackId the id of the admin feedback to update
+ * @param userId the id of the admin
+ * @param comment the feedback comment to insert
+ * @param rating the feedback rating to insert (0 for thumbs down, 1 for thumbs up)
+ * @returns ok(IAdminFeedbackSchema) if successfully inserted
+ * @returns err(MissingAdminFeedbackError) if feedback document with the same feedbackId and userId is not found
+ * @returns err(DatabaseError) on database error
+ */
+export const updateAdminFeedback = ({
+  feedbackId,
+  userId,
+  comment,
+  rating,
+}: {
+  feedbackId: string
+  userId: string
+  comment?: string
+  rating?: number
+}) => {
+  return ResultAsync.fromPromise(
+    AdminFeedbackModel.findById(feedbackId),
```

Comment:
I believe this is not needed, since findByIdAndUpdate should be able to return error when document is not found?

Reply:
But yeah I think you are right, I probably can remove the second call to `findByIdAndUpdate` and instead use `update` followed by `save`. Not sure if there are any iops saving here though

Reply:
This is because the flow of the feedback process is like going to be:
1. Make the feedback doc with rating only
2. Update the feedback doc with comments

As seen in the figma [here](https://www.figma.com/file/CMq0CJtawn7NGM7Zdpi8Ti/Form-Design-Master?type=design&node-id=17430-181452&t=n2QYeoXab9QfORnr-0)!

But for extensibility purpose I generalised the update method to take in rating if required as well, hence, kept it as a 'general' method that will patch if body is passed in

Reply:
Have simplified the logic to only one db call as discussed! [4b7ae59](https://github.com/opengovsg/FormSG/pull/6454/commits/4b7ae59ea2a71f87c2f38ecdde3ceff04adf56cd)